provider/asdf/resolve.go

```diff
@@ -6,25 +6,31 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/bitrise-io/bitrise/v2/log"
 	"github.com/bitrise-io/toolprovider/provider"
 	"github.com/hashicorp/go-version"
 )
 
 type ErrNoMatchingVersion struct {
+	RequestedVersion string
 	AvailableVersions []string
 }
 
 func (e ErrNoMatchingVersion) Error() string {
 	if len(e.AvailableVersions) == 0 {
-		return "no matching version found"
+		return "no match for requested version " + e.RequestedVersion
 	}
 
 	versionList := ""
 	for _, v := range e.AvailableVersions {
-		versionList += fmt.Sprintf("- %s\n", v)
+		if strings.HasPrefix(v, e.RequestedVersion) {
+			versionList += fmt.Sprintf("- %s\n", v)
+		}
+	}
+	if versionList == "" {
+		return fmt.Sprintf("no match for requested version %s", e.RequestedVersion)
+	} else {
+		return fmt.Sprintf("no match for %s. Similar versions: \n%s", e.RequestedVersion, versionList)
 	}
-	return fmt.Sprintf("no matching version found, available versions: \n%s", versionList)
```

Comment:
I realized this wasn't practical at all. In practice, tools like Ruby or Node.js has ~1000 available versions, we can't list every single one.